### PR TITLE
Support namespace

### PR DESCRIPTION
**Pull request by Lxxyx**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ export default function (Vue, {
...
     beforeCreate() {
       this.$persist = (names, storeName = defaultStoreName, storeExpiration = defaultExpiration) => {
+
+        let prefix = this.$options.name;
+        if (!prefix) {
+          console.error('You must set name to Vue components while using vue-persist');
+          return;
+        }
+
+        storeName = `${prefix}__${storeName}`
```
Comment (egoist) [inline src/index.js]: why do we need namespace?

File: src/index.js [hunk 2]
```diff
@@ -35,10 +44,13 @@ export default function (Vue, {
...
-              store.data[name] = val
-              write(storeName, JSON.stringify(store))
-            }, { deep: true })
+            this.$watch(name, {
+              handler: val => {
+                store.data[name] = val;
+                write(storeName, JSON.stringify(store));
+              },
+              deep: true
+            })
```
Comment (egoist) [inline src/index.js]: this is incorrect, `handler` is for component option `watch` instead of the `$watch` method.

File: src/index.js (hunk 1 above)
Comment (Lxxyx) [reply src/index.js]: If you use vue-persist in two components with same parameters.
The second component persist data will overwrite the first one.
So we need namespace.

Comment (egoist) [reply src/index.js]: In that case you should use a different store name https://github.com/egoist/vue-persist#name-1

**Pull request closed without merging**